sdk/account_test.go

```diff
@@ -60,82 +63,65 @@ const accountInfoJson = `
    }
 }
 `
+)
 
-func TestAccountService_GetAccountInfo(t *testing.T) {
-	cl, mux, _, teardown, err := setupMockServer()
-	if err != nil {
-		t.Errorf("Account.GetAccountInfo error setting up mock server: %v", err)
-	}
-	defer teardown()
+var (
+	testAddress1 = "SAONSOGFZZHNEIBRYXHDTDTBR2YSAXKTITRFHG2Y"
+	testAddress2 = "SBJ5D7TFIJWPY56JBEX32MUWI5RU6KVKZYITQ2HA"
+	publicKey1   = "27F6BEF9A7F75E33AE2EB2EBA10EF1D6BEA4D30EBD5E39AF8EE06E96E11AE2A9"
+)
 
-	mux.HandleFunc("/account/SAONSOGFZZHNEIBRYXHDTDTBR2YSAXKTITRFHG2Y", func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprint(w, accountInfoJson)
+func TestAccountService_GetAccountInfo(t *testing.T) {
+	mockServer.addRouter(&router{
+		path:     fmt.Sprintf("/account/%s", testAddress1),
+		respBody: accountInfoJson,
 	})
 
-	acc, _, err := cl.Account.GetAccountInfo(context.Background(), &Address{MijinTest, "SAONSOGFZZHNEIBRYXHDTDTBR2YSAXKTITRFHG2Y"})
-	if err != nil {
-		t.Errorf("Account.GetAccountInfo returned error: %s", err)
-	}
+	acc, resp, err := accountClient.GetAccountInfo(context.Background(), &Address{MijinTest, testAddress1})
 
-	if !reflect.DeepEqual(acc, account) {
-		t.Errorf("Account.GetAccountInfo returned %s, want %s", acc, account)
-	}
+	assert.Nilf(t, err, "AccountService.GetAccountInfo returned error: %s", err)
+	validateResponse(t, resp)
+	assert.Equal(t, account.String(), acc.String())
```

Comment:
Imho, do a comparison of structures through its self method in unit-test not quite right, compare entirely through a method 'assert.Equal' or individual fields if they are references